### DISCUSSION TITLE
Establish TCP/TLS on demand

### DISCUSSION THREAD

**Discussion by SerhiiTukalo** [radsecproxy/radsecproxy#158] [General]
Hello!  
We are using radsecproxy on our fleet of Wifi6 Access points, with the forecast of 200k hotspots by the end of the year.  
It's hard to keep established tunnels on server side for such amount of peers. 
Is there a way to establish TCP/TLS on demand, when RADIUS message coming for particular Realm.
Thank you in advance!

**Comment by rpaffrath** [comment]
Hi,

yes, dynamic server discovery like radsecproxy might doing that.

But the realm provider needs to set up a so called naptr/srv record.
For example:

host -t naptr dfn.de
dfn.de has NAPTR record 100 10 "s" "x-eduroam:radius.tls" "" _radsec._tcp.eduroam.de.
and
 host -t srv _radsec._tcp.eduroam.de.
_radsec._tcp.eduroam.de has SRV record 0 10 2083 tld2.eduroam.de.
_radsec._tcp.eduroam.de has SRV record 0 0 2083 tld1.eduroam.de.
_radsec._tcp.eduroam.de has SRV record 0 20 2083 tld3.eduroam.de.

These connection will be dynamically established for a while until they are not needed anymore.

Best regards!
> On 10. Oct 2024, at 12:11, skudetta ***@***.***> wrote:
> 
> 
> Hello!
> We are using radsecproxy on our fleet of Wifi6 Access points, with the forecast of 200k hotspots by the end of the year.
> It's hard to keep established tunnels on server side for such amount of peers.
> Is there a way to establish TCP/TLS on demand, when RADIUS message coming for particular Realm.
> Thank you in advance!
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

--
Security, Trust & Identity Services

Dipl. Inform. Ralf Paffrath
Service Manager for eduroam, easyroam, eduVPN at DFN

E-Mail: ***@***.***, ***@***.***, ***@***.*** | Fon: +49 30 884299-9121/9120 | Fax: 030 88 42 99 370
__________________________________________________________________________________

DFN - Deutsches Forschungsnetz | German National Research and Education Network
Verein zur Förderung eines Deutschen Forschungsnetzes e.V.
Alexanderplatz 1 | 10178 Berlin
https://www.dfn.de

Vorstand: Prof. Dr.-Ing. Stefan Wesner | Prof. Dr. Helmut Reiser | Christian Zens
Geschäftsführung: Dr. Christian Grimm | Jochem Pattloch
VR AG Charlottenburg 7729B | USt.-ID. DE 136623822
